---
title: "R Class Proejct"
output: github_document
---

Chinook baseline metadata and one with the Chinook baseline genotypes. It will be a nice exercise to join them together and then, like the FRH dataset there will be tons of individuals (8000+), lots of metadata fields and genotypes

2 files:
1. baseline_genotypes.csv has 2 column formatted genotypes
2. baseline_metadata.csv has the corresponding metadata

The metadata parsing issues for our practice dataset can be fixed by setting the guess_max to a higher value.  The code below works:
```{r}
library(tidyverse)
genos <- read_csv("data/baseline_genotypes.csv")
meta <- read_csv("data/baseline_metadata.csv", guess_max = 2000)
```

Goals:

* Squish genotypes and metadata together (and document any genotypes without metadata and vice versa)
* Making sure no more than 2 alleles for each SNP
* Check for duplicate genotypes
* Check for duplicate sample names with different genotypes
* Check for date weirdness
* Length vs. weight to check for possible mis-entered data
* Summary stats: # individuals per population, allele frequencies per population
* Check species ID assay for non-Chinook
* Filter based on missing data
* Make a map of sample locations